pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1009,9 +1009,17 @@ func (l *LibvirtDomainManager) preStartHook(vmi *v1.VirtualMachineInstance, doma
 		return domain, fmt.Errorf("creating service account disk failed: %v", err)
 	}
 
-	// set drivers cache mode
 	for i := range domain.Spec.Devices.Disks {
-		err := api.SetDriverCacheMode(&domain.Spec.Devices.Disks[i])
+		// Set missing image format. For some live images (host disk, fs pvc...)
+		// it is convenient to check image formats on the pre-start hook as it
+		// might fail if VM is already running and the image is already in use.
+		err := api.SetFileDriverType(&domain.Spec.Devices.Disks[i], GetImageInfo)
```

Comment:
In general we have a converter context which allows passing in the disk type to the libvirt converter. Consider using that. You can find an example at https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/virtwrap/manager.go#L818 where we do this for the containerDisk. This allows us to keep all translations inside the converter.

Reply:
Yes, I was aware of the converter context on purpose. `qemu-img` fails if the image is already in use as non backing store on every synchronization other than the first startup of the VM. This is the reason I used the pre-start hook, as it only runs on initial first startup. I try to explain this in the comment above, so maybe i need to re-write so that it is better understood.

Reply:
ok, I still think that we should know at this stage already what we are dealing with. We can extract and save the information somewhere else after a first successful extraction.